app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/DeviceContainer.java

```diff
@@ -0,0 +1,201 @@
+/*  Copyright (C) 2015-2018 Taavi Eomäe
+
+    This file is part of Gadgetbridge.
+
+    Gadgetbridge is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Gadgetbridge is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+
+package nodomain.freeyourgadget.gadgetbridge.service;
+
+import java.util.Random;
+
+import nodomain.freeyourgadget.gadgetbridge.externalevents.AlarmClockReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.AlarmReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.BluetoothConnectReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.BluetoothPairingRequestReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.CMWeatherReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.CalendarReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.MusicPlaybackReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.OmniJawsObserver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.PebbleReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.PhoneCallReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.SMSReceiver;
+import nodomain.freeyourgadget.gadgetbridge.externalevents.TimeChangeReceiver;
+import nodomain.freeyourgadget.gadgetbridge.impl.GBDevice;
+
+public class DeviceContainer {
+    private GBDevice GBDevice;
+    private boolean started = false;
+    private DeviceSupport deviceSupport;
+
+    private PhoneCallReceiver phoneCallReceiver = null;
+    private SMSReceiver SMSReceiver = null;
+    private PebbleReceiver pebbleReceiver = null;
+    private MusicPlaybackReceiver musicPlaybackReceiver = null;
+    private TimeChangeReceiver timeChangeReceiver = null;
+    private BluetoothConnectReceiver blueToothConnectReceiver = null;
+    private BluetoothPairingRequestReceiver blueToothPairingRequestReceiver = null;
+    private AlarmClockReceiver alarmClockReceiver = null;
+
+    private AlarmReceiver alarmReceiver = null;
+    private CalendarReceiver calendarReceiver = null;
+    private CMWeatherReceiver CMWeatherReceiver = null;
+    private OmniJawsObserver omniJawsObserver = null;
+    private Random random = new Random();
```

Comment:
Uh, do we really need to have one instance of all these per connected device?

Reply:
If the receivers aren't configured per-device somehow then most likely not.

Reply:
Though it will complicate the code quite a bit when one of the devices is disconnected then it has to be decided if the listener is needed any longer based on all the devices that are connected at that point?

Reply:
All the receivers is one part of the code I haven't been able to verify working either, as mentioned I simply don't have multiple devices that support all the features.